packages/passport/sdk/src/authManager.ts

```diff
@@ -436,15 +438,22 @@ export default class AuthManager {
   }
 
   public async logout(): Promise<void> {
-    return withPassportError<void>(
-      async () => {
-        if (this.logoutMode === 'silent') {
-          return this.userManager.signoutSilent();
-        }
-        return this.userManager.signoutRedirect();
-      },
-      PassportErrorType.LOGOUT_ERROR,
-    );
+    return withPassportError<void>(async () => {
+      const user = await this.userManager.getUser();
+      if (!user) {
+        return;
+      }
+
+      if (user.refresh_token) {
+        await this.userManager.revokeTokens(['refresh_token']);
+      }
+
+      if (this.logoutMode === 'silent') {
+        await this.userManager.signoutSilent();
+      } else {
+        await this.userManager.signoutRedirect();
+      }
```

Comment:
We could improve performance a bit by parallelising the requests when `signoutSilent` is being used, e.g:
```suggestion
      if (this.logoutMode === 'silent') {
        await Promise.all([
          this.userManager.revokeTokens(['refresh_token']),
          this.userManager.signoutSilent(),
        ]);
      } else {
        await this.userManager.revokeTokens(['refresh_token']);
        await this.userManager.signoutRedirect();
      }
```